motoman_driver/src/io_relay.cpp

```diff
@@ -106,19 +106,21 @@ bool MotomanIORelay::readSingleIoCB(
   motoman_msgs::ReadSingleIO::Response &res)
 {
   shared_int io_val= -1;
+  std::string err_msg;
 
   // send message and release mutex as soon as possible
   this->mutex_.lock();
-  bool result = io_ctrl_.readSingleIO(req.address, io_val);
+  bool result = io_ctrl_.readSingleIO(req.address, io_val, err_msg);
   this->mutex_.unlock();
 
   if (!result)
   {
     res.success = false;
 
-    // provide caller with failure indication (not very informative yet)
+    // provide caller with failure indication
+    // TODO: should we also return the result code?
```

Comment:
This would require extending the message.

Reply:
Do you mean providing something more than stating the result code in the error message?

Reply:
Yes. We could add the code itself in a `uint32` field fi.

That would make more automated handling of errors possible. The current message (with a `str` result message) is nice for humans, not so much for machines.

Reply:
Let's do this in a follow-up PR.